include/TPP/Passes.td

```diff
@@ -197,7 +197,9 @@ def TileConsumerAndFuseProducers : Pass<"tile-consumer-and-fuse-producers",
   }];
   let constructor = "mlir::tpp::createTileConsumerAndFuseProducersPass()";
   let options = [
-    ListOption<"tileSizes", "tile-sizes", "int64_t", "Tile sizes">
+    ListOption<"tileSizes", "tile-sizes", "int64_t", "Tile sizes">,
+    Option<"maxDepth", "max-depth", "int64_t", "5", 
```

Comment:
Why do you need a max-depth and why is this a compiler flag? How does that help with the generic case where the pass goes through the whole IR?

@adam-smnk how does this look with the default pipeline?

Reply:
I use the flag to control how many producers I pull in into the fusion. For the pipeline, we can have a greedy strategy.

Reply:
In the default pipeline, we could configure it explicitly to have no limit or it could be better to set it to greedy as the default pass option value.

Reply:
Makes sense, thanks!